### ISSUE TITLE
Fusion Chart events in iOS does not emit. Only the first will emit.

### ISSUE THREAD

**Issue by YeshanJay** [fusioncharts/react-native-fusioncharts#48]
Chart `events` seems to be not working properly. Only the first key in the event object (key-value paired) will fire/emit.

In the example below, `console.log("bI");` will be logged, but not any other event. Not even `onInitialized` since it depends on the `renderComplete` event.

But if we comment out `beforeInitialize` event, the next event would work and so on.
`onInitialized` event will only work if we don't send the `events` prop or pass an empty object.

#### Update
Only occurs in iOS, and works fine in Android.

#### Update 2
Check point 2 in [this blog](https://medium.com/capriza-engineering/communicating-between-react-native-and-the-webview-ac14b8b8b91a#ac6a), under "3. onMessage/postMessage".

>The problem is that doing consecutive window.postMessage calls in iOS will result in losing your first calls as the url is overridden each time before the message is delivered so the last call wins

<br />

### Scenario 1
```js
// myChart.js
...
import ReactNativeFusionCharts from "react-native-fusioncharts";


const _libraryPath = Platform.select({
    android: { uri: "file:///android_asset/fusioncharts.html" },
    ios: require("./../../../assets/fusioncharts.html")
});

export default class MyChart extends PureComponent {
    refChart = null;
    _chartApi = null;

    render() {
        return (
            <ReactNativeFusionCharts
                ref={(ref) => {
                    this.refChart = ref;
                }}
                .... // <-- Add necessary props
                libraryPath={_libraryPath}

                onInitialized={(chartApi) => {
                    console.log("onInitialized - API");    // <-- Not emitting
                    this._chartApi = chartApi;
                }}
                events={{
                    beforeInitialize: () => {
                        console.log("bI");     // <-- Will emit
                    },
                    initialized: () => {
                        console.log("i");    // <-- Not emitting
                    },
                    chartClick: () => {
                         console.log("cclk");    // <-- Not emitting
                    }
                }}
            />
        );
    }

}
```

### Scenario 2
```js
// myChart.js
...
render() {
    return (
        <ReactNativeFusionCharts
            ref={(ref) => {
                this.refChart = ref;
            }}
            .... // <-- Add necessary props
            libraryPath={_libraryPath}

            onInitialized={(chartApi) => {
                console.log("onInitialized - API");    // <-- Not emitting
                this._chartApi = chartApi;
            }}
            events={{
                // beforeInitialize: () => {
                //     console.log("bI");
                // },
                // initialized: () => {
                //     console.log("i");
                // },
                chartClick: () => {
                     console.log("cclk");    // <-- Will emit
                }
            }}
        />
    );
}
...
```

Version:
`react-native`: 0.59.4
`react-native-webview`: 5.8.1
`fushionchart`: 3.13.5
`react-native-fusioncharts`: 3.0.0

**Comment by ndrishinski** [comment]
I am also having an issue only with IOS Events. I have a dragable 2d area chart: 
```
this.state = {
...,

events: {
        dataplotDragEnd: (e, a) => {
            Alert.alert("called")
            console.log(e.data.endValue)
          },
      }
}

<FusionCharts
                            type={this.state.type}
                            width={this.state.width}
                            height={this.state.height}
                            dataFormat={this.state.dataFormat}
                            dataSource={this.state.dataSource}
                            events={this.state.events}
                            libraryPath={this.libraryPath}
                        />
```
It worked as expected on Android. Any solutions for this?

**Comment by rohanoid5** [comment]
@YeshanJay will investigate this issue and give you an update.

**Comment by YeshanJay** [comment]
@rohanoid5 Thank you. Were you able to find the root cause for this issue?

**Comment by YeshanJay** [comment]
Check point 2 in [this blog](https://medium.com/capriza-engineering/communicating-between-react-native-and-the-webview-ac14b8b8b91a#ac6a), under "3. onMessage/postMessage".

>The problem is that doing consecutive window.postMessage calls in iOS will result in losing your first calls as the url is overridden each time before the message is delivered so the last call wins

<br />
Added an update on this issue.

**Comment by rohanoid5** [comment]
Fixed in version [4.1.0](https://github.com/fusioncharts/react-native-fusioncharts/releases/tag/4.1.0)

**Comment by rohanoid5** [comment]
Issue fixed hence closing. Please check the release logs.
